public/pages/DetectorsList/Containers/ConfirmActionModals/ConfirmActionModals.tsx

```diff
@@ -0,0 +1,405 @@
+            fill
+            isLoading={isLoading || props.isListLoading}
+            onClick={async () => {
+              setIsLoading(true);
+              props.onStartDetectors();
+              props.hideModal();
+            }}
+          >
+            {'Start detectors'}
+          </EuiButton>
+        </EuiModalFooter>
+      </EuiModal>
+    </EuiOverlayMask>
+  );
+};
+
+export const ConfirmStopDetectorsModal = (
+  props: ConfirmStopDetectorsModalProps
+) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={props.hideModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            {'Are you sure you want to stop the selected detectors?'}&nbsp;
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <EuiCallOut
+            title="The following detectors will be stopped. Any associated monitors will
+           not be able to receive any anomaly results to generate alerts:"
+            color="warning"
+            iconType="alert"
+          ></EuiCallOut>
+          <EuiSpacer size="m" />
+          <div>
+            {props.isListLoading ? (
+              <EuiLoadingSpinner size="xl" />
+            ) : (
+              getNamesAndMonitorsGrid(props.detectors, props.monitors)
+            )}
+          </div>
+        </EuiModalBody>
+        <EuiModalFooter>
+          <EuiButtonEmpty
+            data-test-subj="cancelButton"
+            onClick={props.hideModal}
+          >
+            Cancel
+          </EuiButtonEmpty>
+          <EuiButton
+            data-test-subj="confirmButton"
+            color="primary"
+            fill
+            isLoading={isLoading || props.isListLoading}
+            onClick={async () => {
+              setIsLoading(true);
+              props.onStopDetectors();
+              props.hideModal();
+            }}
+          >
+            {'Stop detectors'}
+          </EuiButton>
+        </EuiModalFooter>
+      </EuiModal>
+    </EuiOverlayMask>
+  );
+};
+
+export const ConfirmDeleteDetectorsModal = (
+  props: ConfirmDeleteDetectorsModalProps
+) => {
+  const containsEnabled = containsEnabledDetectors(props.detectors);
+  const [deleteTyped, setDeleteTyped] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={props.hideModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            {'Are you sure you want to delete the selected detectors?'}&nbsp;
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          {containsEnabled ? (
+            <EuiCallOut
+              title="Some of the selected detectors are currently running"
+              color="warning"
+              iconType="alert"
+            ></EuiCallOut>
+          ) : null}
+          {containsEnabled ? <EuiSpacer size="s" /> : null}
```

Comment:
maybe combine those 2 into 1 `{containsEnabled? ...: null}`

Reply:
thanks for noticing, but I can't since JSX expressions can only have one parent element.

Reply:
what about combining them into `<div>...</div>`? just a small thing. feel free to skip it.

Reply:
good idea, that's cleaner. will change